lib/bridge/IrcHandler.js

```diff
@@ -176,6 +176,66 @@ IrcHandler.prototype.onPrivateMessage = Promise.coroutine(function*(req, server,
     yield this.ircBridge.sendMatrixAction(pmRoom, virtualMatrixUser, mxAction, req);
 });
 
+
+/**
+ * Called when the AS receives an IRC topic event.
+ * @param {IrcServer} server : The sending IRC server.
+ * @param {IrcUser} fromUser : The sender.
+ * @param {string} channel : The target channel.
+ * @param {Object} action : The IRC action performed.
+ * @return {Promise} which is resolved/rejected when the request finishes.
+ */
+IrcHandler.prototype.onTopic = Promise.coroutine(function*(req, server, fromUser,
+                                                    channel, action) {
+    req.log.info("onTopic: %s from=%s to=%s action=%s",
+        server.domain, fromUser, channel, JSON.stringify(action).substring(0, 80)
+    );
+
+    // Only bridge topics for rooms created by the bridge, via !join or an alias
+    let origins = ["join", "alias"];
+    let mappings = yield this.ircBridge.getStore().getMappingsForChannelByOrigin(
+        server, channel, origins
+    );
+    if (mappings.length === 0) {
+        req.log.info(
+            "No mapped matrix rooms for IRC channel %s with origin = [%s]",
+            channel,
+            origins
+        );
+        return Promise.resolve();
+    }
+    let promises = mappings.map((entry) => {
+        if (entry.matrix.topic === action.text) {
+            req.log.info(
+                "Topic of %s already set to '%s'",
+                entry.matrix.getId(),
+                action.text
+            );
+            return;
+        }
+        entry.matrix.topic = action.text;
+        this.ircBridge.getStore().upsertRoomStoreEntry(entry);
+        req.log.info(
+            "Message is a topic, using AS bot to set topic in room %s: %s",
+            entry.matrix.getId(),
+            action.text
```

Comment:
The main use for these log lines is for debugging when things blow up. You want to minimise the amount of noise in the logs. Most of this line is redundant and not telling you anything: ""Message is a topic, using AS bot to set topic in room" has little information, and even the params given at the end are missing one of the most important pieces of information: which channel? I would say:
```js
`New topic in ${channel} - bot setting topic in ${entry.matrix.getId()}`
```

We don't care about the topic content.